parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -58,6 +58,28 @@ write_files:
     {{GetVariableProperty "cloudInitData" "initAKSCustomCloud"}}
 {{end}}
 
+{{if and IsHostedMaster EnableHostsConfigAgent}}
+- path: /opt/azure/containers/reconcilePrivateHosts.sh
+  permissions: "0744"
+  encoding: gzip
+  owner: root
+  content: !!binary |
+    {{GetVariableProperty "cloudInitData" "reconcilePrivateHosts"}}
+
+- path: /etc/systemd/system/reconcile-private-hosts.service
+  permissions: "0644"
+  owner: root
+  content: |
+    [Unit]
+    Description=Reconcile /etc/hosts file for private cluster
+    [Service]
+    Type=simple
+    Restart=on-failure
+    ExecStart=/bin/bash /opt/azure/containers/reconcilePrivateHosts.sh
+    [Install]
+    WantedBy=multi-user.target
```

Comment:
can we move this into a file an encode it with gzip? 
cloud init data has size limit, we need to minimize the payload as  much as we can

Reply:
fixed.